keps/sig-cluster-lifecycle/kubeadm/20191229-kubeadm-move-out-of-kk.md

```diff
@@ -0,0 +1,635 @@
+latest Git tags and SHAs.
+
+#### PR testing
+
+There is already support for running pre-submit jobs on k/kubeadm PRs.
+Once kubeadm is moved to the new location a collection of PR blocking
+jobs should be added to make sure the project is always buildable with
+the current version of Go that the k/k repository uses. Syncing Go versions
+should be the responsibility of the k/kubeadm maintainers.
+
+The new PR blocking jobs must support building kubeadm from source with
+the latest commits that a PR for k/kubeadm introduces. The rest of the
+k8s artifacts can be pulled from the CI GCS bucket. This can also allow
+catching problems introduced in k/k when using kubeadm as the deployer.
+
+The /hack/verify-* and update-* scripts that k/kubeadm/kubeadm uses
+will not be a direct copy of k/k but will provide similar results:
+Go versions are in sync Go-fmt, linter and import tools are in sync
+as much as possible
+
+The following lists of tests will be executed on PRs as blocking:
+- Unit tests
+- Integration tests (currently reside in k/k/cmd/kubeadm/test)
+- E2e tests from the e2e-kubeadm suite (currently resides
+in k/k/test/e2e-kubeadm)
+- Full cluster deployment e2e, possibly upgrade and skew tests too
+
+#### Periodics jobs
+
+No changes to the current kubeadm periodic jobs will be required
+as long as the CI build process continues to push the kubeadm
+and the rest of the k8s binaries to the GCS buckets.
+
+### Graduation Criteria
+
+Not applicable as the work in this KEP should be ideally executed within
+one k8s release cycle.
+
+### Upgrade / Downgrade Strategy
+
+No changes to the upgrade strategy will be required.
+Downgrades are not supported by kubeadm.
+
+### Version Skew Strategy
+
+No changes will be applied to the current version skew support in kubeadm
+as long as kubeadm preserves its exiting versioning, which matches
+the versioning of k8s.
+
+## Implementation History
+
+- 2019-12-29: This proposal document was created
+
+## Open Questions
+
+#### Can we stop building DEBs and RPMs from k/k?
+
+Currently the Kubernetes CI build process uses Bazel specs from the
+k/k `/build` directory to build DEBs and RPMs. Packages are generated
+for kubelet, kubeadm as the main targets. The Bazel specs assume kubeadm
+exists under `/cmd/kubeadm` so this process has to be either changed
+or completely removed.
+
+Arguments for the removal of this process from k/k:
+
+- The existence of the CI artifacts is not documented in kubeadm
+and kubelet documentation.
+- Observations by triaging issues in k/k and k/kubeadm in the past
+couple of years have revealed no users of the CI DEBs and RPMs.
+- DEBs and RPMs are also managed in k/release thus the k/k package
+specs can end up out of date.
+
+This KEP proposes the that k/k DEB and RPM build process is completely
+removed from k/k and that it is consolidated over at the k/release
+repository, where there is ongoing work to refactor the package building.
+
+## Drawbacks
+
+NONE
```

Comment:
The branching/tagging scheme complexity and potential end-user confusion points and mis-synchronization feel like a drawback worth mentioning, even if described already in detail above.

Reply:
> potential end-user confusion

potential user confusion can come only if we stop versioning k/kubeadm the same as k/k.

> mis-synchronization

this is already outlined in risks and mitigations.

if SIG Release has other drawbacks please clarify.

Reply:
ok, i'm going to add some comments related to post-submit failures and the human factor of miss communication.